### ISSUE TITLE
Odd issue with EF7 (RC2) returning all virtual collections from entity, without the use of Include

### ISSUE THREAD

**Issue by trustieee** [dotnet/efcore#5461]
It looks like Entity Framework 7, while lazy loading isn't supported / implemented, is returning all relationships for a given entity without using the `Include(...)` method.

I'm not sure I follow what the expected behavior should be in this case. See my reproduction below. Note, [this github branch/commit](https://github.com/discord-csharp/pouarf/tree/ef7-repo) reproduces this.

Add `DbContext` and a repository provider:

```
// In ConfigureServices
services.AddDbContext<PouarfDbContext>(options => options.UseInMemoryDatabase());
services.AddScoped<IContactProvider, ContactProvider>();
```

Seed application data:

```
// In Configure
// CreateSampleData() adds a bunch of `People` objects to a list, 
//  and then loops through them, adding each one to the DbContext 
//  through the `ContactProvider`
Task.Run(async () => 
  await new 
  MockPeople(app
      .ApplicationServices
      .GetService<IContactProvider>()).CreateSampleData());
```

Couple of examples from `ContactProvider` implementation:

```
public async Task AddPerson(Person person)
{
    await Task.Run(() => _dbContext.Add(person));
}

public async Task<IEnumerable<Person>> GetPeople()
{
    // Notice no Include(...)
    return await _dbContext.People.ToListAsync();
}
```

Person object:

```
public class Person : ContactInformationBase // Contains Id
{
    ...

    public virtual ICollection<PhoneNumber> PhoneNumbers { get; set; }
}
```

**And then** in my controller action, I'm getting the people from EF:

```
public async Task<IActionResult> Index()
{
    // Contains a populated list of `PhoneNumbers`, even though
    //  they should only be included if I explicitly asked EF7 to load them.
    var people = await _contactProvider.GetPeople();
    ..
}
```

So basically, EF7 is providing mapped entities from my main object, even though they should only be included when I explicitly ask for them, via an `Include(p => p.PhoneNumbers)`.

I'm using ASP.NET Core 1 RC2, with Entity Framework 7

**EDIT**

Just added an API action to my controller, and the results are... odd.

```
[Route("api/[action]")]
public async Task<IEnumerable<Person>> People()
{
    // True adds all the Includes off the DbSet as needed.
    return await _contactProvider.GetPeople(true);
}
```

Now I'm not getting back any mapped entities with this call. If I remove or add `Include()` call, it doesn't have an effect on the result. Only the core `Person` entity is returned, all mapped entities are `null`.

**Comment by mariocatch** [comment]
Solved.

The issue was in the approach of utilizing the `Include` pattern on the DbSet.

I had this, **WRONG**:

            var people = _dbContext.People;

            if (includeRelationships)
            {
                people.Include(p => p.PhoneNumbers)
                    .Include(p => p.EmailAddresses)
                    .Include(p => p.StreetAddresses);
            }

            return await people.FirstOrDefaultAsync(p => p.Id.Equals(id));

And changed it to this, notice the actual applying of the `Include` here, **RIGHT**:

            IQueryable<Person> people = _dbContext.People;
            if (includeRelationships)
            {
                people = people
                    .Include(p => p.EmailAddresses)
                    .Include(p => p.PhoneNumbers)
                    .Include(p => p.StreetAddresses);
            }

            return await people.FirstOrDefaultAsync(p => p.Id.Equals(id));
